core/services/keystore/eth.go

```diff
@@ -32,22 +33,22 @@ type Eth interface {
 	Disable(address common.Address, chainID *big.Int, qopts ...pg.QOpt) error
 	Reset(address common.Address, chainID *big.Int, nonce int64, qopts ...pg.QOpt) error
 
-	NextSequence(address common.Address, chainID *big.Int, qopts ...pg.QOpt) (int64, error)
-	IncrementNextSequence(address common.Address, chainID *big.Int, currentNonce int64, qopts ...pg.QOpt) error
+	NextSequence(address *evmtypes.Address, chainID *big.Int, qopts ...pg.QOpt) (int64, error)
+	IncrementNextSequence(address *evmtypes.Address, chainID *big.Int, currentNonce int64, qopts ...pg.QOpt) error
 
 	EnsureKeys(chainIDs ...*big.Int) error
 	SubscribeToKeyChanges() (ch chan struct{}, unsub func())
 
-	SignTx(fromAddress common.Address, tx *types.Transaction, chainID *big.Int) (*types.Transaction, error)
+	SignTx(fromAddress *evmtypes.Address, tx *types.Transaction, chainID *big.Int) (*types.Transaction, error)
```

Comment:
I reckon all of these can be value types rather than pointers, removes the need for nil checks then

Reply:
There is a comment thread somewhere but basically when your generic type needs _mutation_ methods, those have to be defined on pointer receivers, along with the rest of the method set. Without generics, you can use the value type and call methods defined on the value or pointer without having to think about it either way, because the `value.prtMethod()`->`(&value).ptrMethod()`is implicit.

Reply:
@prashantkumar1982 IIRC there was a comment about trying to remove some unnecessary mutation methods, was that for a follow up?

Reply:
@prashantkumar1982 IIRC there was a comment about trying to remove some unnecessary mutation methods, was that for a follow up?

Reply:
Oh, `UnmarshalText(text []byte) error` is used. So an alternative would be to pass a separate, equivalent constructor so that the interface (w/o UnmarshalText) could be implemented by a value instead of a pointer. Seems like it could be a messy cascading change though :shrug: WDYT @prashantkumar1982

Reply:
I don't even need UnmarshallText anywhere. It was just added to confirm to golang pattern, as you pointed to in an early comment.
I am in favor of making the Hashable interface read-only, so we can get rid of pointers everywhere. But since it's effect will be widespread, I will prefer to do it in another PR.